### DISCUSSION TITLE
flutter build

### DISCUSSION THREAD

**Discussion by junddao** [codemagic-ci-cd/codemagic-docs#2555] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I use latest version connectivity_plus package (v6.0.3)
If downgrade this package, device_info, package_info packages also have to do that...

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/29177302/86751857-0cef-4fa4-873a-02150b45aead)

### Expected results

build success.

### Actual results

failed

**Comment by himesh-cm** [comment]
Hey @junddao 

According to the error information you shared, your workflow requires Java 17.
Please check [this](https://github.com/orgs/codemagic-ci-cd/discussions/2567) post on setting the required version.

Let me know if you are still facing any issues.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
